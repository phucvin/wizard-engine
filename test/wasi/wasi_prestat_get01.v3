layout Prestat {
    +0   tag: u32;
    +4   dir_len: u32;
    =8;
}

var prestatbuf = Array<byte>.new(8);
var prestatbuf_ref = Ref<Prestat>.of(prestatbuf);

def main() -> int {
    def prestatbuf_p = Pointer.atContents(prestatbuf);
    // A preopen directory will be opened just next to stdout/stdin/stderr, which is 3
    def errno = wasi_snapshot_preview1.fd_prestat_get(3, prestatbuf_p);
    if (errno != wasi_errno.SUCCESS.code) {
        return 1;
    }

    if (prestatbuf_ref.tag != 0) {
        return 2;
    }
    // "preopen_dir" is specified as a preopen directory in flag file
    // so, dir_len will be 11
    if (prestatbuf_ref.dir_len != 11) {
        return 3;
    }

    return 0;
}
