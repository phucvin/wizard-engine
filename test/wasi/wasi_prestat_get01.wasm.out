wasi_snapshot_preview1.fd_prestat_get(3, 16)
  ==> 0
